src/core/grid_based_algorithms/lbgpu.cpp

```diff
@@ -102,9 +100,9 @@ bool ek_initialized = false;
 /*-----------------------------------------------------------*/
 
 /** %Lattice Boltzmann update gpu called from integrate.cpp */
-void lattice_boltzmann_update_gpu() {
+void lattice_boltzmann_update_gpu(double time_step) {
 
-  auto factor = (int)round(lbpar_gpu.tau / time_step);
+  auto factor = static_cast<int>(round(lbpar_gpu.tau / time_step));
```

Comment:
the decision to update the LB should be made in the integration loop and not within the LB. Then the time step is not needed here

Reply:
Thanks! This is now fixed.